app/utils/visualizations/histogram.py

```diff
@@ -5,6 +5,68 @@
 # Imports
 import plotly.express as px
 
+'''
+The following functions serve as validtion on the HistogramRequest
+class outlined in app/api/models.py. They test for input specifications
+that are specific to Story Squad; such as the grade level being between
+8-12.
+'''
```

Comment:
I agree with adding the edge cases to this file for testing but I do not think they need to be in function format. You did a lot of work here and it looks great but the actual testing logic is what we are writing in the test file. If you want to get with me, I can explain my thoughts on this in more detail. This was a great idea though if we did not have the testing file set up.

Reply:
Thank you for clarifying why you added the functions in this way.